modules/auxiliary/gather/ldap_query.rb

```diff
@@ -263,6 +269,36 @@ def output_data_csv(entries)
     generate_rex_tables(entries, 'csv')
   end
 
+  def find_root_dn(ldap)
+    filter = '(objectClass=*)'
+    attributes = ['instanceType']
+    @root_dn = @base_dn
+    loop do
+      attributes_data = perform_ldap_query(ldap, filter, attributes, base: @root_dn)
+      for entry in attributes_data do
+        int_instance_type = entry[:instancetype][0].to_i
+        # Check if this is a root of a naming context (NC).
+        if (int_instance_type & IT_NC_HEAD)
+          # Check if this has a naming context above this one
+          if (int_instance_type & IT_NC_ABOVE)
+            # If it does then set the @root_dn value to that and try again to see if there
+            # are entries higher in the tree.
+            parts = @root_dn.split(',')
+            if parts.length <= 2
+              print_good("Discovered root DN: #{@root_dn}")
+              return @root_dn
+            end
+            @root_dn = parts[1..].join(',')
+          else
+            # We have found the root naming context (NC)
+            print_good("Discovered root DN: #{@root_dn}")
+            @root_dn
```

Comment:
This should be returning here to break out of the loop. Should likely update this to be `return @root_dn`

Reply:
Removed this and loop and the constants with the revised update.